### PR TITLE
Add mutation for forgotten password

### LINKED ISSUE #176: Add mutation for forgotten password
- Mutation should accept user email as argument
- Send POST request to `https://auth.skypicker.com/v1/user.resetPassword` 
- payload should be `{ login: 'userEmail@example.com' }`
- Authorization: `Basic ${config.auth.basicToken}` needs to be included in HTTP headers

### PR DESCRIPTION
**Pull request by radkag**
Closes #176

### CODE REVIEW

File: src/identity/mutations/__tests__/ResetPassword.INVALID_XXX.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  it('INVALID_ARGUMENT_AUTH fail should handle backend gracefully', async () => {
+    RestApiMock.onPost(config.restApiEndpoint.resetPassword).replyWithData({
+      error_code: 'INVALID_ARGUMENT_AUTH',
+      message: 'No authentication credentials.',
+    });
+
+    expect(await graphql(loginQuery)).toMatchSnapshot();
+  });
+
+  it('INVALID_ARGUMENT_LOGIN fail should handle backend gracefully', async () => {
```
Comment (radkag) [inline src/identity/mutations/__tests__/ResetPassword.INVALID_XXX.test.js]: I used `LOGIN`, cause I wan't sure what errors are actually coming from backend.

File: src/identity/mutations/__tests__/Login.INVALID_XXX.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// @flow
+
```
Comment (radkag) [inline src/identity/mutations/__tests__/Login.INVALID_XXX.test.js]: Is there idea how to better call this file? :)))

File: src/identity/User.js [hunk 3]
```diff
@@ -1,5 +1,9 @@
 // @flow
 
+export type ForgottenPasswordResponse = {|
```
Comment (jaroslav-kubicek) [inline src/identity/User.js]: since it's named as `ResetPassword`, I would name this reset as well

Comment (jaroslav-kubicek) [reply src/identity/User.js]: also move it to `src/identity/mutations/ResetPassword.js` because it's used only there if I'm correct

File: src/identity/mutations/__tests__/Login.INVALID_XXX.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+// @flow
```
Comment (jaroslav-kubicek) [inline src/identity/mutations/__tests__/Login.INVALID_XXX.test.js]: It's definitely nice you are refactoring tests for `Login` as well, but that belongs to separate PR, sorry

File: src/identity/mutations/__tests__/ResetPassword.INVALID_XXX.test.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    RestApiMock.onPost(config.restApiEndpoint.resetPassword).replyWithData({
+      error_code: 'INVALID_ARGUMENT_LOGIN',
+      message: 'Invalid email address provided.',
+      userId: undefined,
```
Comment (jaroslav-kubicek) [inline src/identity/mutations/__tests__/ResetPassword.INVALID_XXX.test.js]: this  `userId: undefined` is really returned from API?

File: src/identity/mutations/__tests__/ResetPassword.INVALID_XXX.test.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    });
+
+    expect(await graphql(loginQuery)).toMatchSnapshot();
+  });
+
+  it('INVALID_LOGIN fail should handle backend gracefully', async () => {
+    RestApiMock.onPost(config.restApiEndpoint.resetPassword).replyWithData({
+      error_code: 'INVALID_LOGIN',
+      message: 'Login failed.',
```
Comment (jaroslav-kubicek) [inline src/identity/mutations/__tests__/ResetPassword.INVALID_XXX.test.js]: Can it really return `Login failed` for reset password?

File: src/identity/types/outputs/ForgottenPasswordResponse.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// @flow
+
+import { GraphQLObjectType, GraphQLBoolean } from 'graphql';
+
+export default new GraphQLObjectType({
+  name: 'ForgottenPasswordResponse',
```
Comment (jaroslav-kubicek) [inline src/identity/types/outputs/ForgottenPasswordResponse.js]: rename to `ResetPasswordResponse` + filename

File: src/identity/mutations/__tests__/Login.INVALID_XXX.test.js (hunk 4 above)
Comment (radkag) [reply src/identity/mutations/__tests__/Login.INVALID_XXX.test.js]: Is it really essential? What's the point of it? :D

File: src/identity/mutations/__tests__/Login.INVALID_XXX.test.js (hunk 2 above)
Comment (jaroslav-kubicek) [reply src/identity/mutations/__tests__/Login.INVALID_XXX.test.js]: filename is imho fine, just new PR

File: src/identity/mutations/__tests__/Login.INVALID_XXX.test.js (hunk 4 above)
Comment (jaroslav-kubicek) [reply src/identity/mutations/__tests__/Login.INVALID_XXX.test.js]: it's not *really essential* but it's more about teaching you good habits - PR should be small and contained.

Comment (mrtnzlml) [reply src/identity/mutations/__tests__/Login.INVALID_XXX.test.js]: Oh yes, I do around 100 code reviews per month and I really wish every single one was just a one-liner. The reality is that almost every single one is thousand-liner and it's really hard to focus and read everything properly... 🙂

Comment (radkag) [reply src/identity/mutations/__tests__/Login.INVALID_XXX.test.js]: All right then. :)) My habits will thank you later!

File: src/identity/types/outputs/ForgottenPasswordResponse.js (hunk 7 above)
Comment (radkag) [reply src/identity/types/outputs/ForgottenPasswordResponse.js]: You mean `ResetPasswordResponseForgottenPasswordResponse`? 😃Nooo!

File: src/identity/mutations/__tests__/ResetPassword.INVALID_XXX.test.js (hunk 6 above)
Comment (radkag) [reply src/identity/mutations/__tests__/ResetPassword.INVALID_XXX.test.js]: How to find out the all possible error messages from backend?

File: src/identity/types/outputs/ForgottenPasswordResponse.js (hunk 7 above)
Comment (jaroslav-kubicek) [reply src/identity/types/outputs/ForgottenPasswordResponse.js]: wut? I mean the mutation is called `resetPassword`, so let's rename this file to `ResetPasswordResponse.js` + change the name to `ResetPasswordResponse`

**Comment by mrtnzlml** [comment]
@radkag Please rebase it on top of the master branch so we can review it again.

File: config/application.js [hunk 8]
```diff
@@ -19,6 +19,7 @@ export default {
         v: 3,
       }),
     login: 'https://auth.skypicker.com/v1/user.login',
+    resetPassword: 'https://auth.skypicker.com/v1/user.resetPassword',
```
Comment (mrtnzlml) [inline config/application.js]: Please try to avoid adding new lines to this file.

Comment (radkag) [reply config/application.js]: Could you share the reason with me? :))

Comment (mrtnzlml) [reply config/application.js]: This file is deprecated.

Comment (jaroslav-kubicek) [reply config/application.js]: wut? 😄  (although I can probably understand why) it's new to me...

Comment (michalsanger) [reply config/application.js]: @jaroslav-kubicek @radkag since we are doing feature based folder structure, what's the reason of this file? To have all URLs on one place? Do we need that? Only DataLoader(s) needs that, so let's put it there.

**Comment by radkag** [comment]
All fixed.
